shared_model/backend/protobuf/impl/proto_query_response_factory.cpp

```diff
@@ -68,6 +68,29 @@ shared_model::proto::ProtoQueryResponseFactory::createAccountAssetResponse(
       query_hash);
 }
 
+std::unique_ptr<shared_model::interface::QueryResponse>
+shared_model::proto::ProtoQueryResponseFactory::createAccountAssetResponse(
+    const std::vector<interface::types::AccountIdType> &account_ids,
+    const std::vector<interface::types::AssetIdType> &asset_ids,
+    const std::vector<shared_model::interface::Amount> &balances,
+    const crypto::Hash &query_hash) const {
+  return createQueryResponse(
+      [account_ids = std::move(account_ids),
+       asset_ids = std::move(asset_ids),
+       balances = std::move(balances)](
+          iroha::protocol::QueryResponse &protocol_query_response) {
+        iroha::protocol::AccountAssetResponse *protocol_specific_response =
+            protocol_query_response.mutable_account_assets_response();
+        for (decltype(account_ids.size()) i = 0; i < account_ids.size(); i++) {
+          auto *asset = protocol_specific_response->add_account_assets();
+          asset->set_account_id(account_ids.at(i));
```

Comment:
I am not sure, but won't it be useful, if you `std::move(account_ids.at(i))` and all other values?